src/bun.js/node/node_crypto_binding.zig

```diff
@@ -17,56 +17,249 @@ const JSError = bun.JSError;
 const String = bun.String;
 const UUID = bun.UUID;
 
-fn randomInt(globalThis: *JSC.JSGlobalObject, callframe: *JSC.CallFrame) bun.JSError!JSC.JSValue {
-    const arguments = callframe.arguments_old(2).slice();
+const random = struct {
+    const max_possible_length = @min(JSC.ArrayBuffer.max_size, std.math.maxInt(i32));
+    const max_range = 0xffff_ffff_ffff;
 
-    //min, max
-    if (!arguments[0].isNumber()) return globalThis.throwInvalidArgumentTypeValue("min", "safe integer", arguments[0]);
-    if (!arguments[1].isNumber()) return globalThis.throwInvalidArgumentTypeValue("max", "safe integer", arguments[1]);
-    const min = arguments[0].to(i64);
-    const max = arguments[1].to(i64);
+    fn randomInt(global: *JSC.JSGlobalObject, callFrame: *JSC.CallFrame) bun.JSError!JSC.JSValue {
+        const args = callFrame.arguments();
 
-    if (min > JSC.MAX_SAFE_INTEGER or min < JSC.MIN_SAFE_INTEGER) {
-        return globalThis.throwInvalidArgumentRangeValue("min", "It must be a safe integer type number", min);
+        const min_value, const max_value, const callback, const min_specified = args: {
+            if (args.len == 2) {
+                break :args .{ JSC.jsNumber(0), args[0], args[1], false };
+            }
+
+            break :args .{ args[0], args[1], args[2], true };
+        };
+
+        if (!callback.isUndefined()) {
+            _ = try validators.validateFunction(global, "callback", callback);
+        }
+
+        if (!min_value.isSafeInteger()) {
+            return global.throwInvalidArgumentTypeValue2("min", "a safe integer", min_value);
+        }
+        if (!max_value.isSafeInteger()) {
+            return global.throwInvalidArgumentTypeValue2("max", "a safe integer", max_value);
+        }
+
+        const min: i64 = @intFromFloat(@trunc(min_value.asNumber()));
+        const max: i64 = @intFromFloat(@trunc(max_value.asNumber()));
```

Comment:
```suggestion
    const min = try validators.validateInteger(global, min_value, "min", null, null);
    const max = try validators.validateInteger(global, max_value, "max", null, null);
```

Reply:
node uses `Number.isSafeInteger` on these values instead of using a validator